Add recode road 1.0

hulsiejames opened the pull request:
Code for the `recode_road_class` function created in respose to #65

Robinlovelace commented on test-code/recode_road_class.R:
```diff
@@ -0,0 +1,50 @@
+
+# Define recode_road_class function
+recode_road_class <- function(osm_sf) {
+  library("dplyr")
```
This is pretty unconventional. For package code you don't need to attach the whole namespace. Your use of `dplyr::*()` means you don't need to load the package.

hulsiejames replied:
I see - I had confused myself thinking you needed to specify ask some functions can be masked by other packages! will remove

Robinlovelace commented on test-code/recode_road_class.R:
```diff
@@ -0,0 +1,50 @@
+# Date      27/06/2022
```
I suggest moving this file into the R directory.

hulsiejames replied:
Which directory is this? 
`openinfra/code` ?

Robinlovelace reviewed (changes requested):
Minor suggested changes in comments.

hulsiejames reviewed (commented):
removed `library(dplyr)` from fucntion code. 
Not sure which directory the R directory is

Robinlovelace commented:
Just create a new folder, e.g. with `dir.create("R")`. Details...

Robinlovelace commented:
You could try `usethis::use_package()`

Robinlovelace commented:
Or better, `usethis::use_r()`: https://r-pkgs.org/whole-game.html#use_r

Robinlovelace commented:
Build is failing as there's no NAMESPACE. I can talk you over how to create that: https://r-pkgs.org/namespace.html

hulsiejames commented:
As can be seen above I have had some issues - documenting here. 

- [2a596b8](https://github.com/udsleeds/openinfra/pull/67/commits/2a596b8e195406d483ba2d8f69360930dd7a8096) failed on roxygen example as ` Error in recode_road_class(total_place = sf::read_sf("https://github.com/udsleeds/openinfra/releases/download/v0.2/bbbike_leeds_27_6_22.geojson")) : 
    could not find function "recode_road_class"`. To fix this I added `openinfra::recode_road`

- [f1d748e](https://github.com/udsleeds/openinfra/pull/67/commits/f1d748e890a80a7ffbe973ed4421c16e9c9fd4b7) error due to function not being exported ` Error: 'recode_road_class' is not an exported object from 'namespace:openinfra'`. info was NOT in the NAMESPACE file as I was missing an additional Roxygen comment (#' @export). once I added this I re-ran devtools::document() and the NAMESPACE file was updated

- [fa63fc2](https://github.com/udsleeds/openinfra/pull/67/commits/fa63fc212046f1ffc1fb15dbd0cc96c4b36cd353) this workflow failed due to :` unused argument (total_place = sf::read_sf("https://github.com/udsleeds/openinfra/releases/download/v0.2/bbbike_leeds_27_6_22.geojson"))` To try and fix this I simply removed `total_place=` so that no value is assigned and the sf object is passed to `openinfra::recode_road_class`

- [dbec52d](https://github.com/udsleeds/openinfra/pull/67/commits/dbec52d64ee7847dcf28092366de10d135f401c2) workflow run failed on: `Error in osm_sf %>% dplyr::mutate(road_class = dplyr::case_when(highway %in%  : 
    could not find function "%>%"` %>% is not base R and is part of dplyr or maggritr - to try and fix this error I added `dplyr::` before every `%>%` however using `devtools::document()` resultsed in an error in `parse(text = lines, n = -1, srcfile = srcfile)`, specifically this error listed `36:   osm_recat = osm_sf dplyr`, pointing to the d in dplyr. Instead I opted to change %>% to base pipe `|>`. This seems to have worked...

- Finally, [71645e5](https://github.com/udsleeds/openinfra/pull/67/commits/71645e5b6be75050c2507d90a38764d750bc5bb1) has failed running example: `output = openinfra::recode_road_class(sf::read_sf("https://github.com/udsleeds/openinfra/releases/download/v0.2/bbbike_leeds_27_6_22.geojson"))`  Specifically: `Error in data.matrix(data) : 'list' object cannot be coerced to type 'double'`

Not too sure on final error, not the expected behaviour, going to play around with it now and figure out what causes the error

Robinlovelace commented:
Also: what do you get when running `devtools::check()`?

hulsiejames commented:
> Also: what do you get when running `devtools::check()`?

<details>
```
> devtools::check()
ℹ Updating openinfra documentation
ℹ Loading openinfra
Warning: 
── Conflicts ────────────────────────────────────────────────────────────────────────────────────────────────── openinfra conflicts ──
✖ recode_road_class() masks openinfra::recode_road_class()

Did you accidentally source a file rather than using `load_all()`?
Run `rm(list = c("recode_road_class"))` to remove the conflicts.
── Building ───────────────────────────────────────────────────────────────────────────────────────────────────────────── openinfra ──
Setting env vars:
• CFLAGS    : -Wall -pedantic -fdiagnostics-color=always
• CXXFLAGS  : -Wall -pedantic -fdiagnostics-color=always
• CXX11FLAGS: -Wall -pedantic -fdiagnostics-color=always
• CXX14FLAGS: -Wall -pedantic -fdiagnostics-color=always
• CXX17FLAGS: -Wall -pedantic -fdiagnostics-color=always
• CXX20FLAGS: -Wall -pedantic -fdiagnostics-color=always
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
✔  checking for file ‘/home/james/Desktop/LIDA_OSM_Project/openinfra/openinfra/DESCRIPTION’ ...
─  preparing ‘openinfra’:
✔  checking DESCRIPTION meta-information ...
─  installing the package to build vignettes
E  creating vignettes (15.5s)
   --- re-building ‘QGIS_OSM_Setup.Rmd’ using rmarkdown
   --- finished re-building ‘QGIS_OSM_Setup.Rmd’
   
   --- re-building ‘getosm.Rmd’ using rmarkdown
   ── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──
   ✔ ggplot2 3.3.6     ✔ purrr   0.3.4
   ✔ tibble  3.1.7     ✔ dplyr   1.0.9
   ✔ tidyr   1.2.0     ✔ stringr 1.4.0
   ✔ readr   2.1.2     ✔ forcats 0.5.1
   ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
   ✖ dplyr::filter() masks stats::filter()
   ✖ dplyr::lag()    masks stats::lag()
   Quitting from lines 36-56 (getosm.Rmd) 
   Error: processing vignette 'getosm.Rmd' failed with diagnostics:
   there is no package called 'kableExtra'
   --- failed re-building ‘getosm.Rmd’
   
   --- re-building ‘im_get.Rmd’ using rmarkdown
   --- finished re-building ‘im_get.Rmd’
   
   --- re-building ‘openinfra.Rmd’ using rmarkdown
   Quitting from lines 92-99 (openinfra.Rmd) 
   Error: processing vignette 'openinfra.Rmd' failed with diagnostics:
   there is no package called 'kableExtra'
   --- failed re-building ‘openinfra.Rmd’
   
   --- re-building ‘osm-python.Rmd’ using rmarkdown
   --- finished re-building ‘osm-python.Rmd’
   
   --- re-building ‘regions.Rmd’ using rmarkdown
   --- finished re-building ‘regions.Rmd’
   
   SUMMARY: processing the following files failed:
     ‘getosm.Rmd’ ‘openinfra.Rmd’
   
   Error: Vignette re-building failed.
   In addition: Warning message:
   In tools::buildVignettes(dir = ".", tangle = TRUE) :
     Files named as vignettes but with no recognized vignette engine:
      ‘vignettes/JOSIS_article.Rmd’
      ‘vignettes/data_packs.Rmd’
      ‘vignettes/package_networks.Rmd’
   (Is a VignetteBuilder field missing?)
   Execution halted
Error in c("(function (command = NULL, args = character(), error_on_status = TRUE, ",  : 
  ! System command 'R' failed
Type .Last.error.trace to see where the error occurred
```
</details>

Robinlovelace commented:
Try to get the tests passing locally is my suggestion. Any questions on how to do that welcome.

hulsiejames commented:
New `devtools:check()` FULL run in details below. 

Summary: 2 errors ✖ | 2 warnings ✖ | 6 notes ✖

2 Errors: 
- portable file names - easy fix and not breaking
- checking examples error, the same one i describe a couple comments up `'list' object cannot be coerced to type 'double'`

2 Warnings: 
- `qpdf` is needed for checks on size reduction of PDFs (no issue, installing now)
- another warning about portable file names 

**Just the results**
```
── R CMD check results ───────────────────────────────────────────────────────────────────────────────────── openinfra 0.0.0.9000 ────
Duration: 2m 22.8s

❯ checking for portable file names ... ERROR
  Found the following file with a non-portable file name:
    % after using dplyr::%>% I had devtools:document() error in parse lines >dplyr::%>% 
  These are not valid file names on all R platforms.
  Please rename the files and try again.
  See section ‘Package structure’ in the ‘Writing R Extensions’ manual.
   WARNING
  Found the following file with a non-portable file name:
    % after using dplyr::%>% I had devtools:document() error in parse lines >dplyr::%>% 
  These are not fully portable file names.
  See section ‘Package structure’ in the ‘Writing R Extensions’ manual.
  Found the following non-portable file paths:
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap-icons.css
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap-icons.woff
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap.min.css
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap.min.js
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/clipboard/clipboard.min.js
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/anchor.min.js
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/popper.min.js
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/quarto-syntax-highlighting.css
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/tippy.umd.min.js
  
  Tarballs are only required to store paths of up to 100 bytes and cannot
  store those of more than 256 bytes, with restrictions including to 100
  bytes for the final component.
  See section ‘Package structure’ in the ‘Writing R Extensions’ manual.

❯ checking examples ... ERROR
  Running examples in ‘openinfra-Ex.R’ failed
  The error most likely occurred in:
  
  > base::assign(".ptime", proc.time(), pos = "CheckExEnv")
  > ### Name: recode_road_class
  > ### Title: Re-classifies OSM Roads
  > ### Aliases: recode_road_class
  > 
  > ### ** Examples
  > 
  > output = recode_road_class(sf::read_sf("https://github.com/udsleeds/openinfra/releases/download/v0.2/bbbike_leeds_27_6_22.geojson"))
  Error in data.matrix(data) : 
    'list' object cannot be coerced to type 'double'
  Calls: recode_road_class ... filter -> as.ts -> as.ts.default -> ts -> data.matrix
  Execution halted

❯ checking for portable file names ... ERROR
  Found the following file with a non-portable file name:
    % after using dplyr::%>% I had devtools:document() error in parse lines >dplyr::%>% 
  These are not valid file names on all R platforms.
  Please rename the files and try again.
  See section ‘Package structure’ in the ‘Writing R Extensions’ manual.
   WARNING
  Found the following file with a non-portable file name:
    % after using dplyr::%>% I had devtools:document() error in parse lines >dplyr::%>% 
  These are not fully portable file names.
  See section ‘Package structure’ in the ‘Writing R Extensions’ manual.
  Found the following non-portable file paths:
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap-icons.css
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap-icons.woff
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap.min.css
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap.min.js
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/clipboard/clipboard.min.js
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/anchor.min.js
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/popper.min.js
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/quarto-syntax-highlighting.css
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/tippy.umd.min.js
  
  Tarballs are only required to store paths of up to 100 bytes and cannot
  store those of more than 256 bytes, with restrictions including to 100
  bytes for the final component.
  See section ‘Package structure’ in the ‘Writing R Extensions’ manual.

❯ checking for unstated dependencies in examples ... OK
   WARNING
  ‘qpdf’ is needed for checks on size reduction of PDFs

❯ checking installed package size ... NOTE
    installed size is 30.0Mb
    sub-directories of 1Mb or more:
      doc  29.9Mb

❯ checking top-level files ... NOTE
  Non-standard files/directories found at top level:
    ‘% after using dplyr::%>% I had devtools:document() error in parse
    lines >dplyr::%>% ’ ‘JOSIS’ ‘Python_scripts’ ‘Quarto_docs’
    ‘ResCompLeedsCon’ ‘lida_report’ ‘modelling_world_2022’
    ‘presentations’ ‘references.bib’ ‘setup.R’ ‘sotm2022’

❯ checking dependencies in R code ... NOTE
  Namespaces in Imports field not imported from:
    ‘od’ ‘osmextract’ ‘pct’ ‘sf’ ‘stats19’ ‘tmap’
    All declared Imports should be used.

❯ checking R code for possible problems ... NOTE
  recode_road_class: no visible global function definition for ‘filter’
  recode_road_class: no visible binding for global variable ‘road_class’
  Undefined global functions or variables:
    filter road_class
  Consider adding
    importFrom("stats", "filter")
  to your NAMESPACE file.

❯ checking Rd line widths ... NOTE
  Rd file 'recode_road_class.Rd':
    \examples lines wider than 100 characters:
       output = recode_road_class(sf::read_sf("https://github.com/udsleeds/openinfra/releases/download/v0.2/bbbike_leeds_27_6_22.geojson"))
  
  These lines will be truncated in the PDF manual.

❯ checking files in ‘vignettes’ ... NOTE
  Files named as vignettes but with no recognized vignette engine:
     ‘vignettes/JOSIS_article.Rmd’
     ‘vignettes/data_packs.Rmd’
     ‘vignettes/package_networks.Rmd’
  (Is a VignetteBuilder field missing?)

2 errors ✖ | 2 warnings ✖ | 6 notes ✖
```

<details>

```

> devtools::check()
ℹ Updating openinfra documentation
ℹ Loading openinfra
── Building ───────────────────────────────────────────────────────────────────────────────────────────────────────────── openinfra ──
Setting env vars:
• CFLAGS    : -Wall -pedantic -fdiagnostics-color=always
• CXXFLAGS  : -Wall -pedantic -fdiagnostics-color=always
• CXX11FLAGS: -Wall -pedantic -fdiagnostics-color=always
• CXX14FLAGS: -Wall -pedantic -fdiagnostics-color=always
• CXX17FLAGS: -Wall -pedantic -fdiagnostics-color=always
• CXX20FLAGS: -Wall -pedantic -fdiagnostics-color=always
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
✔  checking for file ‘/home/james/Desktop/LIDA_OSM_Project/openinfra/openinfra/DESCRIPTION’ ...
─  preparing ‘openinfra’:
✔  checking DESCRIPTION meta-information ...
─  installing the package to build vignettes
✔  creating vignettes (2m 1.8s)
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘openinfra_0.0.0.9000.tar.gz’
   Warning in utils::tar(filepath, pkgname, compression = compression, compression_level = 9L,  :
     storing paths of more than 100 bytes is not portable:
     ‘openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap-icons.css’
   Warning in utils::tar(filepath, pkgname, compression = compression, compression_level = 9L,  :
     storing paths of more than 100 bytes is not portable:
     ‘openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap-icons.woff’
   Warning in utils::tar(filepath, pkgname, compression = compression, compression_level = 9L,  :
     storing paths of more than 100 bytes is not portable:
     ‘openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap.min.css’
   Warning in utils::tar(filepath, pkgname, compression = compression, compression_level = 9L,  :
     storing paths of more than 100 bytes is not portable:
     ‘openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap.min.js’
   Warning in utils::tar(filepath, pkgname, compression = compression, compression_level = 9L,  :
     storing paths of more than 100 bytes is not portable:
     ‘openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/clipboard/clipboard.min.js’
   Warning in utils::tar(filepath, pkgname, compression = compression, compression_level = 9L,  :
     storing paths of more than 100 bytes is not portable:
     ‘openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/anchor.min.js’
   Warning in utils::tar(filepath, pkgname, compression = compression, compression_level = 9L,  :
     storing paths of more than 100 bytes is not portable:
     ‘openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/popper.min.js’
   Warning in utils::tar(filepath, pkgname, compression = compression, compression_level = 9L,  :
     storing paths of more than 100 bytes is not portable:
     ‘openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/quarto-syntax-highlighting.css’
   Warning in utils::tar(filepath, pkgname, compression = compression, compression_level = 9L,  :
     storing paths of more than 100 bytes is not portable:
     ‘openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/tippy.umd.min.js’
   
── Checking ───────────────────────────────────────────────────────────────────────────────────────────────────────────── openinfra ──
Setting env vars:
• _R_CHECK_CRAN_INCOMING_USE_ASPELL_: TRUE
• _R_CHECK_CRAN_INCOMING_REMOTE_    : FALSE
• _R_CHECK_CRAN_INCOMING_           : FALSE
• _R_CHECK_FORCE_SUGGESTS_          : FALSE
• NOT_CRAN                          : true
── R CMD check ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
─  using log directory ‘/tmp/Rtmp6I4KcK/openinfra.Rcheck’
─  using R version 4.2.1 (2022-06-23)
─  using platform: x86_64-pc-linux-gnu (64-bit)
─  using session charset: UTF-8
─  using options ‘--no-manual --as-cran’ (1.5s)
✔  checking for file ‘openinfra/DESCRIPTION’
─  this is package ‘openinfra’ version ‘0.0.0.9000’
─  package encoding: UTF-8
✔  checking package namespace information
✔  checking package dependencies (1s)
✔  checking if this is a source package ...
✔  checking if there is a namespace
✔  checking for executable files (394ms)
✔  checking for hidden files and directories ...
E  checking for portable file names
   Found the following file with a non-portable file name:
     % after using dplyr::%>% I had devtools:document() error in parse lines >dplyr::%>% 
   These are not valid file names on all R platforms.
   Please rename the files and try again.
   See section ‘Package structure’ in the ‘Writing R Extensions’ manual.
    WARNING
   Found the following file with a non-portable file name:
     % after using dplyr::%>% I had devtools:document() error in parse lines >dplyr::%>% 
   These are not fully portable file names.
   See section ‘Package structure’ in the ‘Writing R Extensions’ manual.
   Found the following non-portable file paths:
     openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap-icons.css
     openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap-icons.woff
     openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap.min.css
     openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap.min.js
     openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/clipboard/clipboard.min.js
     openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/anchor.min.js
     openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/popper.min.js
     openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/quarto-syntax-highlighting.css
     openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/tippy.umd.min.js
   
   Tarballs are only required to store paths of up to 100 bytes and cannot
   store those of more than 256 bytes, with restrictions including to 100
   bytes for the final component.
   See section ‘Package structure’ in the ‘Writing R Extensions’ manual.
✔  checking for sufficient/correct file permissions ...
✔  checking serialization versions
✔  checking whether package ‘openinfra’ can be installed (838ms)
N  checking installed package size ...
     installed size is 30.0Mb
     sub-directories of 1Mb or more:
       doc  29.9Mb
✔  checking package directory
✔  checking for future file timestamps (404ms)
✔  checking ‘build’ directory ...
✔  checking DESCRIPTION meta-information ...
N  checking top-level files ...
   Non-standard files/directories found at top level:
     ‘% after using dplyr::%>% I had devtools:document() error in parse
     lines >dplyr::%>% ’ ‘JOSIS’ ‘Python_scripts’ ‘Quarto_docs’
     ‘ResCompLeedsCon’ ‘lida_report’ ‘modelling_world_2022’
     ‘presentations’ ‘references.bib’ ‘setup.R’ ‘sotm2022’
✔  checking for left-over files
✔  checking index information ...
✔  checking package subdirectories ...
✔  checking R files for non-ASCII characters ...
✔  checking R files for syntax errors ...
✔  checking whether the package can be loaded ...
✔  checking whether the package can be loaded with stated dependencies ...
✔  checking whether the package can be unloaded cleanly ...
✔  checking whether the namespace can be loaded with stated dependencies ...
✔  checking whether the namespace can be unloaded cleanly ...
✔  checking loading without being on the library search path ...
N  checking dependencies in R code (386ms)
   Namespaces in Imports field not imported from:
     ‘od’ ‘osmextract’ ‘pct’ ‘sf’ ‘stats19’ ‘tmap’
     All declared Imports should be used.
✔  checking S3 generic/method consistency ...
✔  checking replacement functions ...
✔  checking foreign function calls ...
N  checking R code for possible problems (1.4s)
   recode_road_class: no visible global function definition for ‘filter’
   recode_road_class: no visible binding for global variable ‘road_class’
   Undefined global functions or variables:
     filter road_class
   Consider adding
     importFrom("stats", "filter")
   to your NAMESPACE file.
✔  checking Rd files ...
✔  checking Rd metadata ...
N  checking Rd line widths ...
   Rd file 'recode_road_class.Rd':
     \examples lines wider than 100 characters:
        output = recode_road_class(sf::read_sf("https://github.com/udsleeds/openinfra/releases/download/v0.2/bbbike_leeds_27_6_22.geojson"))
   
   These lines will be truncated in the PDF manual.
✔  checking Rd cross-references ...
✔  checking for missing documentation entries ...
✔  checking for code/documentation mismatches ...
✔  checking Rd \usage sections (364ms)
✔  checking Rd contents ...
✔  checking for unstated dependencies in examples ...
    WARNING
   ‘qpdf’ is needed for checks on size reduction of PDFs
✔  checking installed files from ‘inst/doc’ ...
N  checking files in ‘vignettes’ ...
   Files named as vignettes but with no recognized vignette engine:
      ‘vignettes/JOSIS_article.Rmd’
      ‘vignettes/data_packs.Rmd’
      ‘vignettes/package_networks.Rmd’
   (Is a VignetteBuilder field missing?)
E  checking examples (24s)
   Running examples in ‘openinfra-Ex.R’ failed
   The error most likely occurred in:
   
   > base::assign(".ptime", proc.time(), pos = "CheckExEnv")
   > ### Name: recode_road_class
   > ### Title: Re-classifies OSM Roads
   > ### Aliases: recode_road_class
   > 
   > ### ** Examples
   > 
   > output = recode_road_class(sf::read_sf("https://github.com/udsleeds/openinfra/releases/download/v0.2/bbbike_leeds_27_6_22.geojson"))
   Error in data.matrix(data) : 
     'list' object cannot be coerced to type 'double'
   Calls: recode_road_class ... filter -> as.ts -> as.ts.default -> ts -> data.matrix
   Execution halted
✔  checking for unstated dependencies in vignettes ...
✔  checking package vignettes in ‘inst/doc’ ...
✔  checking re-building of vignette outputs (1m 48.2s)
✔  checking for non-standard things in the check directory ...
✔  checking for detritus in the temp directory
   
   See
     ‘/tmp/Rtmp6I4KcK/openinfra.Rcheck/00check.log’
   for details.
   
   
── R CMD check results ───────────────────────────────────────────────────────────────────────────────────── openinfra 0.0.0.9000 ────
Duration: 2m 22.8s

❯ checking for portable file names ... ERROR
  Found the following file with a non-portable file name:
    % after using dplyr::%>% I had devtools:document() error in parse lines >dplyr::%>% 
  These are not valid file names on all R platforms.
  Please rename the files and try again.
  See section ‘Package structure’ in the ‘Writing R Extensions’ manual.
   WARNING
  Found the following file with a non-portable file name:
    % after using dplyr::%>% I had devtools:document() error in parse lines >dplyr::%>% 
  These are not fully portable file names.
  See section ‘Package structure’ in the ‘Writing R Extensions’ manual.
  Found the following non-portable file paths:
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap-icons.css
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap-icons.woff
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap.min.css
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap.min.js
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/clipboard/clipboard.min.js
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/anchor.min.js
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/popper.min.js
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/quarto-syntax-highlighting.css
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/tippy.umd.min.js
  
  Tarballs are only required to store paths of up to 100 bytes and cannot
  store those of more than 256 bytes, with restrictions including to 100
  bytes for the final component.
  See section ‘Package structure’ in the ‘Writing R Extensions’ manual.

❯ checking examples ... ERROR
  Running examples in ‘openinfra-Ex.R’ failed
  The error most likely occurred in:
  
  > base::assign(".ptime", proc.time(), pos = "CheckExEnv")
  > ### Name: recode_road_class
  > ### Title: Re-classifies OSM Roads
  > ### Aliases: recode_road_class
  > 
  > ### ** Examples
  > 
  > output = recode_road_class(sf::read_sf("https://github.com/udsleeds/openinfra/releases/download/v0.2/bbbike_leeds_27_6_22.geojson"))
  Error in data.matrix(data) : 
    'list' object cannot be coerced to type 'double'
  Calls: recode_road_class ... filter -> as.ts -> as.ts.default -> ts -> data.matrix
  Execution halted

❯ checking for portable file names ... ERROR
  Found the following file with a non-portable file name:
    % after using dplyr::%>% I had devtools:document() error in parse lines >dplyr::%>% 
  These are not valid file names on all R platforms.
  Please rename the files and try again.
  See section ‘Package structure’ in the ‘Writing R Extensions’ manual.
   WARNING
  Found the following file with a non-portable file name:
    % after using dplyr::%>% I had devtools:document() error in parse lines >dplyr::%>% 
  These are not fully portable file names.
  See section ‘Package structure’ in the ‘Writing R Extensions’ manual.
  Found the following non-portable file paths:
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap-icons.css
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap-icons.woff
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap.min.css
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/bootstrap/bootstrap.min.js
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/clipboard/clipboard.min.js
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/anchor.min.js
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/popper.min.js
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/quarto-syntax-highlighting.css
    openinfra/Quarto_docs/Experiences working with OSM data in Python_files/libs/quarto-html/tippy.umd.min.js
  
  Tarballs are only required to store paths of up to 100 bytes and cannot
  store those of more than 256 bytes, with restrictions including to 100
  bytes for the final component.
  See section ‘Package structure’ in the ‘Writing R Extensions’ manual.

❯ checking for unstated dependencies in examples ... OK
   WARNING
  ‘qpdf’ is needed for checks on size reduction of PDFs

❯ checking installed package size ... NOTE
    installed size is 30.0Mb
    sub-directories of 1Mb or more:
      doc  29.9Mb

❯ checking top-level files ... NOTE
  Non-standard files/directories found at top level:
    ‘% after using dplyr::%>% I had devtools:document() error in parse
    lines >dplyr::%>% ’ ‘JOSIS’ ‘Python_scripts’ ‘Quarto_docs’
    ‘ResCompLeedsCon’ ‘lida_report’ ‘modelling_world_2022’
    ‘presentations’ ‘references.bib’ ‘setup.R’ ‘sotm2022’

❯ checking dependencies in R code ... NOTE
  Namespaces in Imports field not imported from:
    ‘od’ ‘osmextract’ ‘pct’ ‘sf’ ‘stats19’ ‘tmap’
    All declared Imports should be used.

❯ checking R code for possible problems ... NOTE
  recode_road_class: no visible global function definition for ‘filter’
  recode_road_class: no visible binding for global variable ‘road_class’
  Undefined global functions or variables:
    filter road_class
  Consider adding
    importFrom("stats", "filter")
  to your NAMESPACE file.

❯ checking Rd line widths ... NOTE
  Rd file 'recode_road_class.Rd':
    \examples lines wider than 100 characters:
       output = recode_road_class(sf::read_sf("https://github.com/udsleeds/openinfra/releases/download/v0.2/bbbike_leeds_27_6_22.geojson"))
  
  These lines will be truncated in the PDF manual.

❯ checking files in ‘vignettes’ ... NOTE
  Files named as vignettes but with no recognized vignette engine:
     ‘vignettes/JOSIS_article.Rmd’
     ‘vignettes/data_packs.Rmd’
     ‘vignettes/package_networks.Rmd’
  (Is a VignetteBuilder field missing?)

2 errors ✖ | 2 warnings ✖ | 6 notes ✖

```

</details>

hulsiejames commented:
Main confusion comes from: 
```
❯ checking examples ... ERROR
  Running examples in ‘openinfra-Ex.R’ failed
  The error most likely occurred in:
  
  > base::assign(".ptime", proc.time(), pos = "CheckExEnv")
  > ### Name: recode_road_class
  > ### Title: Re-classifies OSM Roads
  > ### Aliases: recode_road_class
  > 
  > ### ** Examples
  > 
  > output = recode_road_class(sf::read_sf("https://github.com/udsleeds/openinfra/releases/download/v0.2/bbbike_leeds_27_6_22.geojson"))
  Error in data.matrix(data) : 
    'list' object cannot be coerced to type 'double'
  Calls: recode_road_class ... filter -> as.ts -> as.ts.default -> ts -> data.matrix
  Execution halted
```

As I have not been able to re-create this error locally and the function works exactly as expected

Robinlovelace commented:
Maybe try adding `usethis::use_pipe()`.

hulsiejames commented:
- [ad76f8a](https://github.com/udsleeds/openinfra/pull/67/commits/ad76f8a5836bbc56a3d8d0d9201eb9fd22c6d853)

^ changed pipe operator from `|>` back to `%>%` with the addition of `usethis::use_pipe()`

hulsiejames commented:
I see - I'm a little confused as I'm not sure where some of these package imports have come from - for example 'od' 'stats19' 'tmap'? 

Also as that is a warning i don't think it is what is failing the workflow, but the error from the example seems to fail it

hulsiejames commented:
All seems to be working well now. :tada:

Accepting the PR.

Issue Documentation
_______________________________________________

Issue: I was not using `dplyr::` before `filter(! is.na(road_class))`

Function de-bugging is more challenging, but, making use of  [`browser()`](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/browser) makes de-bugging much easier
